src/main/scala/no/ndla/searchapi/service/search/IndexService.scala

```diff
@@ -50,27 +50,33 @@ trait IndexService {
       } yield imported
     }
 
-    def indexDocuments(implicit mf: Manifest[D]): Try[ReindexResult] = synchronized {
-      val start = System.currentTimeMillis()
-      createIndexWithGeneratedName.flatMap(indexName => {
-        val operations = for {
-          numIndexed <- sendToElastic(indexName)
-          aliasTarget <- getAliasTarget
-          _ <- updateAliasTarget(aliasTarget, indexName)
-        } yield numIndexed
-
-        operations match {
-          case Failure(f) =>
-            deleteIndexWithName(Some(indexName))
-            Failure(f)
-          case Success(totalIndexed) =>
-            Success(ReindexResult(totalIndexed, System.currentTimeMillis() - start))
-        }
-      })
-    }
+    def indexDocuments(taxonomyBundle: Option[Bundle] = None)(implicit mf: Manifest[D]): Try[ReindexResult] =
```

Comment:
Vi bør prøve å unngå å sende med `Option`s som argumenter til funksjoner. Det kan ofte gjøres bedre ved å bruke pure verdier. `Some` og `None` casene kan i stedet håndteres der de oppstår i stedet.

Her kan man f.eks ha to funksjoner:
`def indexDocuments(taxonomyBundle: Bundle)` og en `def indexDocuments()`. `indexDocuments` kan fetche `taxonomyBundle` først, så kalle `indexDocuments(taxonomyBundle)`.